core/templates/services/promo-bar.service.ts

```diff
@@ -15,31 +15,36 @@
 /**
  * @fileoverview Service Promo bar.
  */
-
-require('services/services.constants.ajs.ts');
-
-angular.module('oppia').factory('PromoBarService', [
-  '$http', '$q', 'ENABLE_PROMO_BAR',
-  function($http, $q, ENABLE_PROMO_BAR) {
-    return {
-      getPromoBarData: function() {
-        var promoBarData = {
-          promoBarEnabled: false,
-          promoBarMessage: ''
-        };
-
-        if (ENABLE_PROMO_BAR) {
-          return $http.get('/promo_bar_handler', {}).then(
-            function(response) {
-              promoBarData.promoBarEnabled = response.data.promo_bar_enabled;
-              promoBarData.promoBarMessage = response.data.promo_bar_message;
-              return promoBarData;
-            }
-          );
-        } else {
-          return $q.resolve(promoBarData);
-        }
-      }
+import { Injectable } from '@angular/core';
+import { HttpClient } from '@angular/common/http';
+import { downgradeInjectable } from '@angular/upgrade/static';
+import { ServicesConstants } from 'services/services.constants';
+@Injectable({
+  providedIn: 'root'
+})
+export class PromoBarService {
+  constructor(
+      private http: HttpClient
+  ) {}
+  getPromoBarData(): Promise<{}> {
+    var promoBarData = {
+      promoBarEnabled: false,
+      promoBarMessage: ''
     };
+    if (ServicesConstants.ENABLE_PROMO_BAR) {
+      return this.http.get('/promo_bar_handler', { observe: 'response' })
```

Comment:
Please make a new backend-api service that makes this HTTP request and use that service here.

Reply:
@nishantwrp 
in which folder is the backend service to be made?
thanks

Reply:
You can create a service named `PromoBarBackendApiService` in `core/templates/services` directory.

Reply:
And it has to be specified in angular-service.ts .?

Reply:
Yes it has to be added in angular-name.service